### DISCUSSION TITLE
QUIC Server Preview Branch Available for Testing and Feedback

### DISCUSSION THREAD

**Discussion by hlandau** [openssl/openssl#24160] [Announcements]
We are pleased to announce the availability of a feature preview for our OpenSSL QUIC server functionality. This is an early technology preview which is being published to seek feedback from our communities.

This preview is now available in the [feature/quic-server branch](https://github.com/openssl/openssl/tree/feature/quic-server) of the OpenSSL repository on GitHub. Those interested in providing early feedback on our QUIC server functionality are invited to download and build this branch.

It is important to note that this branch represents a prototype phase at this time and many aspects of the planned functionality are not yet implemented. In particular, only a very small subset of the full SSL API is currently implemented. This preview is being released to enable all of our communities to provide their feedback as part of the API design process and in order to validate our requirements prior to the finalisation of the API.

This branch is intended as a feature preview to support development and testing. It has not yet been subject to ordinary QA processes, is not yet fully RFC-conformant, and is not intended for production use.

A demo program demonstrating usage of the new server API is [available here](https://github.com/openssl/openssl/tree/feature/quic-server/demos/quic/server), which is the recommended starting point for those seeking to experiment with this feature preview.

Official support for server-side QUIC functionality is anticipated to be delivered in OpenSSL 3.4, which, as per our time-based releases policy, will be released no later than 31 October 2024.

As always, bug reports and issues can be [filed on our issue tracker](https://github.com/openssl/openssl/issues), and questions about using this feature preview can be posted below. Comments on this preview are also welcomed via email to [feedback@openssl.org](mailto:feedback@openssl.org).

**Comment by surpass007** [comment]
hello, I read from https://openssl-library.org/post/2024-03-10-quic-server-preview/   that quic server will be released no later than 31 October 2024. but It seems that the openssl 3.4 haven't integrate quic server yet, I wonder the when quic server will be released, thanks

**Reply by Sashan** [reply]
Hello,

unfortunately the quic server is still in progress. We will appreciate if you try a code from [feature/quic-server](https://github.com/openssl/openssl/tree/feature/quic-server) branch. There is also very simple demo server [there](https://github.com/openssl/openssl/tree/feature/quic-server/demos/quic/server).

thanks and
regards
